### PR TITLE
[Profile] Drop support for old-style managed identity account

### PR DESCRIPTION
**Pull request by jiasli**
**Related command**
`az login`

**Description**
This PR is separated from https://github.com/Azure/azure-cli/pull/25959#discussion_r1151373278

In versions <= 2.0.50 (released on November 6, 2018), `_SUBSCRIPTION_NAME` is used to denote the managed identity ID info (`_ASSIGNED_IDENTITY_INFO`). This is a bad design, as an Azure subscription has its own name. So, `_ASSIGNED_IDENTITY_INFO` was added in https://github.com/Azure/azure-cli/pull/7744 to give way to the real subscription name and `_try_parse_msi_account_name()` acts as an adaptor. However, such logic is difficult to maintain. Even its creator admits:

> the code is a bit messy here to support both old and new styles.
> https://github.com/Azure/azure-cli/pull/7744#discussion_r232437583

As it has been 6 years since 2.0.50, and Azure CLI is self-consistent, this should not be considered a breaking change.

**Testing Guide**
```
az login --identity
az group list
az account get-access-token

az login --identity --username <client_id/object_id/resource_id>
az group list
az account get-access-token
```

**History Notes**

[Profile] Drop support for old-style managed identity account created by Azure CLI <= 2.0.50

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by jiasli** [comment]
CI failed:

https://dev.azure.com/azclitools/public/_build/results?buildId=205660&view=logs&jobId=a8943ac2-38d7-5792-f2a7-5f4fd06db24e&j=a8943ac2-38d7-5792-f2a7-5f4fd06db24e&t=3ed51913-4dd7-564f-a8d8-fda07de13946

```
[Profile] Drop support for old-style managed identity account created by Azure CLI <= 2.0.50. If you upgrade from one of these versions, please run `az login --identity` again.
: missing ` around --identity`
                                                                                                                                                              ↑         ↑
```

I don't think this detection is correct.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -642,15 +642,18 @@ def get_subscription_id(self, subscription=None):  # take id or name
...
-        msi_info, user = account[_USER_ENTITY].get(_ASSIGNED_IDENTITY_INFO), account[_USER_ENTITY].get(_USER_NAME)
-
-        if user in [_SYSTEM_ASSIGNED_IDENTITY, _USER_ASSIGNED_IDENTITY]:
-            if not msi_info:
-                msi_info = account[_SUBSCRIPTION_NAME]  # fall back to old persisting way
-            parts = msi_info.split('-', 1)
-            if parts[0] in MsiAccountTypes.valid_msi_account_types():
-                return parts[0], (None if len(parts) <= 1 else parts[1])
+    def _parse_managed_identity_account(account):
+        user_name = account[_USER_ENTITY][_USER_NAME]
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_profile.py:646]: `_USER_NAME` must exist:

https://github.com/Azure/azure-cli/blob/77f14ba7f5e235a554b4edf1978e14195225f540/src/azure-cli-core/azure/cli/core/_profile.py#L491-L494

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 2]
```diff
@@ -642,15 +642,18 @@ def get_subscription_id(self, subscription=None):  # take id or name
...
+        if user_name == _SYSTEM_ASSIGNED_IDENTITY:
+            # The account contains:
+            #   "assignedIdentityInfo": "MSI",
+            #   "name": "systemAssignedIdentity",
+            return MsiAccountTypes.system_assigned, None
+        if user_name == _USER_ASSIGNED_IDENTITY:
+            # The account contains:
+            #   "assignedIdentityInfo": "MSIClient-xxx"/"MSIObject-xxx"/"MSIResource-xxx",
+            #   "name": "userAssignedIdentity",
+            return tuple(account[_USER_ENTITY][_ASSIGNED_IDENTITY_INFO].split('-', maxsplit=1))
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_profile.py:656]: If `_USER_NAME` is `_SYSTEM_ASSIGNED_IDENTITY` or `_USER_ASSIGNED_IDENTITY`, `_ASSIGNED_IDENTITY_INFO` must exist:

https://github.com/Azure/azure-cli/blob/77f14ba7f5e235a554b4edf1978e14195225f540/src/azure-cli-core/azure/cli/core/_profile.py#L276-L285

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/tests/test_profile.py [hunk 3]
```diff
@@ -1009,22 +1017,17 @@ def test_get_login_credentials_aux_tenants(self, get_user_credential_mock):
...
     @mock.patch('azure.cli.core.auth.adal_authentication.MSIAuthenticationWrapper', MSRestAzureAuthStub)
-    def test_get_login_credentials_msi_system_assigned(self):
-
-        # setup an existing msi subscription
+    def test_get_login_credentials_mi_system_assigned(self):
         profile = Profile(cli_ctx=DummyCli(), storage={'subscriptions': None})
-        test_subscription_id = '12345678-1bf0-4dda-aec3-cb9272f09590'
-        test_tenant_id = '12345678-38d6-4fb2-bad9-b7b93a3e1234'
-        test_user = 'systemAssignedIdentity'
-        msi_subscription = SubscriptionStub('/subscriptions/' + test_subscription_id, 'MSI', self.state1, test_tenant_id)
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/tests/test_profile.py:1019]: Test cases still use subscription name to store the managed identity info.

**Pull request merged**